### ISSUE TITLE
idea : support for UNIX sockets

### ISSUE THREAD

**Issue by jhgoebbert** [ideonate/jhsingle-native-proxy#24]
Hello everyone,  

Running web applications on multi-user systems (even behind a JupyterHub) requires that the web application supports some kind of authentication (this can be a token that has to be passed in the URL, as we know it from Jupyter/JupyterLab).

However, some applications lack this basic security and are accessible to anyone who can log into the multi-user system and access their port.

If these web applications **would not use a port but a UNIX socket** for internal communication with jhsingle-native-proxy, this security hole could be closed in a quite elegant way.  
For jupyter-server-proxy this great feature was just recently added - https://github.com/jupyterhub/jupyter-server-proxy/issues/321 and might be a starting point.

**Comment by danlester** [comment]
Yes, this project was originally a fork of jupyter-server-proxy so it makes sense.

**Comment by danlester** [comment]
Closing due to inactivity automatically.
